libpod/container_internal_linux.go

```diff
@@ -319,6 +320,29 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 	}
 
+	// Add overlay volumes
+	uid := c.RootUID()
+	gid := c.RootGID()
+	idMappings, err := c.IDMappings()
+	if err != nil {
+		return nil, err
+	}
+	if len(idMappings.UIDMap) > 0 && len(idMappings.GIDMap) > 0 {
```

Comment:
Do we need this? I thought RootUID and RootGID already looked through the mappings?

Reply:
Add this because https://github.com/containers/podman/pull/6895#discussion_r451133191, is this the right way to get the uid/gid of the user running in the container?

Reply:
@giuseppe ^^

Reply:
I'd like this answered before merge, otherwise LGTM

Reply:
@giuseppe PTAL

Reply:
yes, I don't think we need the extra check, RootUID/RootGID are enough.

Also it seems to not check the container id but just get the first id available, it can be different than root as the mappings are not ordered